// @dcloudio/uni-cli-nvue/dist/webpack/config/index.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.createConfig = void 0;
const path_1 = __importDefault(require("path"));
const optimization_1 = require("./optimization");
const output_1 = require("./output");
const module_1 = require("./module");
const plugins_1 = require("./plugins");
const resolve_1 = require("./resolve");
const entry_1 = require("./entry");
function createConfig(mode, options) {
    return {
        mode: mode,
        devtool: false,
        watch: mode === 'development',
        entry: (0, entry_1.createEntry)(),
        externals: {
            vue: 'Vue',
        },
        module: (0, module_1.createModule)(options),
        optimization: (0, optimization_1.createOptimization)(),
        output: (0, output_1.createOutput)(),
        resolve: (0, resolve_1.createResolve)(),
        resolveLoader: {
            modules: [
                'node_modules',
                path_1.default.resolve(__dirname, '../../../', 'node_modules'),
            ],
        },
        plugins: (0, plugins_1.createPlugins)(),
    };
}
exports.createConfig = createConfig;
