// @dcloudio/uni-cli-nvue/dist/webpack/config/module/rules/cssLoader/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.createCssLoaders = void 0;
const oneOf_1 = require("./oneOf");
function createCssLoaders() {
    return [
        {
            test: /\.css$/,
            oneOf: (0, oneOf_1.createOneOf)(),
        },
        {
            test: /\.scss$/,
            oneOf: (0, oneOf_1.createOneOf)(scssLoader),
        },
        {
            test: /\.sass$/,
            oneOf: (0, oneOf_1.createOneOf)(sassLoader),
        },
        {
            test: /\.less$/,
            oneOf: (0, oneOf_1.createOneOf)(lessLoader),
        },
        {
            test: /\.styl(us)?$/,
            oneOf: (0, oneOf_1.createOneOf)(stylusLoader),
        },
    ];
}
exports.createCssLoaders = createCssLoaders;
const scssLoader = {
    loader: require.resolve('sass-loader'),
    options: {
        sourceMap: false,
        additionalData: '@import "@/uni.scss";',
        sassOptions: {
            outputStyle: 'expanded',
        },
    },
};
const sassLoader = {
    loader: require.resolve('sass-loader'),
    options: {
        sourceMap: false,
        additionalData: '@import "@/uni.sass"',
        sassOptions: {
            outputStyle: 'expanded',
            indentedSyntax: true,
        },
    },
};
const lessLoader = {
    loader: require.resolve('less-loader'),
    options: {
        sourceMap: false,
    },
};
const stylusLoader = {
    loader: require.resolve('stylus-loader'),
    options: {
        sourceMap: false,
        preferPathResolver: 'webpack',
    },
};
