// @dcloudio/uni-app-vite/dist/nvue/plugins/entry.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.uniNVueEntryPlugin = exports.createUniNVuePagePath = void 0;
const path_1 = __importDefault(require("path"));
const uni_cli_shared_1 = require("@dcloudio/uni-cli-shared");
const uniNVuePagePrefix = 'uniNVuePage://';
function isUniNVuePageUrl(id) {
    return id.startsWith(uniNVuePagePrefix);
}
function createUniNVuePagePath(pagePath) {
    return uniNVuePagePrefix + (0, uni_cli_shared_1.encodeBase64Url)(pagePath + '.nvue');
}
exports.createUniNVuePagePath = createUniNVuePagePath;
function parseUniNvuePagePath(uniNVuePageUrl) {
    return (0, uni_cli_shared_1.decodeBase64Url)(uniNVuePageUrl.replace(uniNVuePagePrefix, ''));
}
function uniNVueEntryPlugin() {
    const inputDir = process.env.UNI_INPUT_DIR;
    return {
        name: 'uni:app-nvue-entry',
        enforce: 'pre',
        resolveId(id) {
            if (isUniNVuePageUrl(id)) {
                return id;
            }
        },
        load(id) {
            if (isUniNVuePageUrl(id)) {
                const filepath = (0, uni_cli_shared_1.normalizePath)(path_1.default.resolve(inputDir, parseUniNvuePagePath(id)));
                this.addWatchFile(filepath);
                return {
                    code: `import { createApp } from 'vue'
import App from '${filepath}'
${uni_cli_shared_1.polyfillCode}
createApp(App).mount('#app')
`,
                };
            }
        },
    };
}
exports.uniNVueEntryPlugin = uniNVueEntryPlugin;
