// @dcloudio/uni-app-vite/dist/nvue/rollup/index.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.createRollupOptions = void 0;
const plugin_vue_1 = __importDefault(require("@vitejs/plugin-vue"));
const uni_cli_shared_1 = require("@dcloudio/uni-cli-shared");
const uni_shared_1 = require("@dcloudio/uni-shared");
const entry_1 = require("../plugins/entry");
function createRollupOptions(pagePath) {
    return {
        input: (0, entry_1.createUniNVuePagePath)(pagePath),
        context: 'global',
        external: ['vue'],
        preserveEntrySignatures: false,
        output: {
            dir: process.env.UNI_OUTPUT_DIR,
            file: (0, uni_cli_shared_1.removeExt)(pagePath) + '.js',
            format: 'iife',
            exports: 'auto',
            sourcemap: false,
            globals: { vue: 'Vue' },
        },
        plugins: [
            (0, plugin_vue_1.default)({
                include: [uni_cli_shared_1.EXTNAME_VUE_RE],
                isProduction: process.env.NODE_ENV === 'production',
                // reactivityTransform: true,
                template: {
                    compilerOptions: {
                        // TODO
                        isNativeTag: uni_shared_1.isBuiltInComponent,
                    },
                },
            }),
        ],
    };
}
exports.createRollupOptions = createRollupOptions;
