// @dcloudio/uni-cli-nvue/dist/webpack/config/module/rules/fileLoader.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.createFileLoader = void 0;
const path_1 = __importDefault(require("path"));
const uni_cli_shared_1 = require("@dcloudio/uni-cli-shared");
function createFileLoader() {
    return {
        test: /\.(png|jpg|gif|ttf|eot|woff|woff2)$/i,
        use: [
            {
                loader: 'file-loader',
                options: {
                    publicPath(_url, resourcePath) {
                        return ('/' +
                            (0, uni_cli_shared_1.normalizePath)(path_1.default.relative(process.env.UNI_INPUT_DIR, resourcePath)));
                    },
                    outputPath(_url, resourcePath) {
                        return (0, uni_cli_shared_1.normalizePath)(path_1.default.relative(process.env.UNI_INPUT_DIR, resourcePath));
                    },
                },
            },
        ],
    };
}
exports.createFileLoader = createFileLoader;
