// @dcloudio/uni-mp-vite/dist/plugins/pagesJson.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.uniPagesJsonPlugin = exports.getNVueCssPaths = void 0;
const fs_1 = __importDefault(require("fs"));
const path_1 = __importDefault(require("path"));
const debug_1 = __importDefault(require("debug"));
const uni_cli_shared_1 = require("@dcloudio/uni-cli-shared");
const entry_1 = require("./entry");
const debugPagesJson = (0, debug_1.default)('uni:pages-json');
const nvueCssPathsCache = new Map();
function getNVueCssPaths(config) {
    return nvueCssPathsCache.get(config);
}
exports.getNVueCssPaths = getNVueCssPaths;
function uniPagesJsonPlugin(options) {
    let resolvedConfig;
    return (0, uni_cli_shared_1.defineUniPagesJsonPlugin)((opts) => {
        return {
            name: 'uni:mp-pages-json',
            enforce: 'pre',
            configResolved(config) {
                resolvedConfig = config;
            },
            transform(code, id) {
                var _a;
                if (!opts.filter(id)) {
                    return null;
                }
                const inputDir = process.env.UNI_INPUT_DIR;
                this.addWatchFile(path_1.default.resolve(inputDir, 'pages.json'));
                (0, uni_cli_shared_1.getLocaleFiles)(path_1.default.resolve(inputDir, 'locale')).forEach((filepath) => {
                    this.addWatchFile(filepath);
                });
                const manifestJson = (0, uni_cli_shared_1.parseManifestJsonOnce)(inputDir);
                const { appJson, pageJsons, nvuePages } = (0, uni_cli_shared_1.parseMiniProgramPagesJson)(code, process.env.UNI_PLATFORM, {
                    debug: !!manifestJson.debug,
                    darkmode: options.app.darkmode &&
                        fs_1.default.existsSync(path_1.default.resolve(inputDir, 'theme.json')),
                    networkTimeout: manifestJson.networkTimeout,
                    subpackages: !!options.app.subpackages,
                    ...options.json,
                });
                nvueCssPathsCache.set(resolvedConfig, nvuePages.map((pagePath) => pagePath + options.style.extname));
                (0, uni_cli_shared_1.mergeMiniProgramAppJson)(appJson, manifestJson[process.env.UNI_PLATFORM]);
                if ((_a = options.json) === null || _a === void 0 ? void 0 : _a.formatAppJson) {
                    options.json.formatAppJson(appJson, manifestJson, pageJsons);
                }
                (0, uni_cli_shared_1.addMiniProgramAppJson)(appJson);
                Object.keys(pageJsons).forEach((name) => {
                    (0, uni_cli_shared_1.addMiniProgramPageJson)(name, pageJsons[name]);
                });
                return {
                    code: `import './manifest.json.js'\n` + importPagesCode(appJson),
                    map: { mappings: '' },
                };
            },
            generateBundle() {
                (0, uni_cli_shared_1.findChangedJsonFiles)().forEach((value, key) => {
                    debugPagesJson('json.changed', key);
                    this.emitFile({
                        type: 'asset',
                        fileName: key + '.json',
                        source: value,
                    });
                });
            },
        };
    });
}
exports.uniPagesJsonPlugin = uniPagesJsonPlugin;
function importPagesCode(pagesJson) {
    const importPagesCode = [];
    function importPageCode(pagePath) {
        const pagePathWithExtname = (0, uni_cli_shared_1.normalizePagePath)(pagePath, process.env.UNI_PLATFORM);
        if (pagePathWithExtname) {
            importPagesCode.push(`import('${(0, entry_1.virtualPagePath)(pagePathWithExtname)}')`);
        }
    }
    pagesJson.pages.forEach((pagePath) => importPageCode(pagePath));
    if (pagesJson.subPackages) {
        pagesJson.subPackages.forEach(({ root, pages }) => {
            pages &&
                pages.forEach((pagePath) => importPageCode(path_1.default.join(root, pagePath)));
        });
    }
    return `if(!Math){
${importPagesCode.join('\n')}
}`;
}
